### PR TITLE
lucene 9.7.0

### PR DESCRIPTION
**Pull request by tarzanek**

### CODE REVIEW

**Comment by tarzanek** [comment]
don't merge until I fix tests

**Comment by tarzanek** [comment]
@ahornace hey, do you have any clue why those tests started failing with my patch?

**Comment by tarzanek** [comment]
hmm, it works when I use deprecated methods, so likely we will need to support them, so let me find a way

**Comment by vladak** [comment]
> hmm, it works when I use deprecated methods, so likely we will need to support them, so let me find a way

What deprecated methods specifically ?

File: suggester/src/main/java/org/opengrok/suggest/SuggesterSearcher.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ public List<LookupResultItem> suggest(
 
         try {
             if (query != null) {
-                rewrittenQuery = query.rewrite(getIndexReader());
+                rewrittenQuery = query.rewrite(this);
```
Comment (tarzanek) [inline suggester/src/main/java/org/opengrok/suggest/SuggesterSearcher.java:108]: because of this line the tests fail ...

**Comment by tarzanek** [comment]
ok, ubuntu works, why isn't mac and windows working? flaky tests?

**Comment by tarzanek** [comment]
fwiw no reindex is needed, this is mostly bugfix

**Comment by vladak** [comment]
@tarzanek is this change ready ?

**Comment by tarzanek** [comment]
yep, feel free to merge @vladak

**Comment by vladak** [comment]
Thanks !

**Pull request merged**